addon/utils/serialize-query-params.ts

```diff
@@ -1,15 +1,17 @@
+import { isPlainObject } from 'ember-fetch/types';
+
 const RBRACKET = /\[\]$/;
 
 /**
  * Helper function that turns the data/body of a request into a query param string.
  * This is directly copied from jQuery.param.
- * @param {Object} queryParamsObject
- * @returns {String}
  */
-export function serializeQueryParams(queryParamsObject) {
-  var s = [];
+export function serializeQueryParams(
+  queryParamsObject: object | string
+): string {
+  var s: any[] = [];
```

Comment:
```suggestion
  const s: any[] = [];
```

Reply:
There're lots of `var` here because it was copy-pasted from jquery code. We can do a refactor in the future and leave this PR a type only one.